### PR TITLE
Add the DeepMind CALM fine-tuning notebook 

### PR DESCRIPTION
**Pull request by kinarr**
- Model composition using 2 Gemma 2 2B models
- Fine-tuning the composed model with a [quotes dataset](https://huggingface.co/datasets/Abirate/english_quotes)
- Colab only and requires an A100 (needs a min of 30GB of VRAM) while Kaggle's T4 x2 runs OOM

### CODE REVIEW

File: Gemma/Finetune_with_CALM.ipynb [hunk 1]
```diff
@@ -0,0 +1,831 @@
+{
```
Comment (windmaple) [inline Gemma/Finetune_with_CALM.ipynb:1]: <p>Wait, CALM is not used to accelerate inference. Its main purpose is to tackle new tasks by re-using existing models. So I'm confused by the 2nd and 3rd paragraph here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/gemma-cookbook/pull/123/discussion/#comment-1889538760'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

Comment (windmaple) [inline Gemma/Finetune_with_CALM.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #8.    elif os.path.exists('/kaggle/working'):
</pre><p>does it work w/ Kaggle? If not, let's remove this</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/gemma-cookbook/pull/123/discussion/#comment-1889538761'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='5'/>

Comment (windmaple) [inline Gemma/Finetune_with_CALM.ipynb:1]: <p>what's 'dynamic early-exit inference'? It's not mentioned in the paper at all</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/gemma-cookbook/pull/123/discussion/#comment-1889538762'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='17'/>

Comment (kinarr) [reply Gemma/Finetune_with_CALM.ipynb:1]: It's mentioned [here](https://research.google/blog/accelerating-text-generation-with-confident-adaptive-language-modeling-calm/).

Comment (kinarr) [reply Gemma/Finetune_with_CALM.ipynb:1]: Will remove it.

Comment (kinarr) [reply Gemma/Finetune_with_CALM.ipynb:1]: It's able to dynamically distribute compute effort across the prediction timesteps so thought I'd include it from the [blog](https://research.google/blog/accelerating-text-generation-with-confident-adaptive-language-modeling-calm/).

Comment (kinarr) [reply Gemma/Finetune_with_CALM.ipynb:1]: I'll remove it.

Comment (kinarr) [reply Gemma/Finetune_with_CALM.ipynb:1]: Will revise it.

Comment (kinarr) [reply Gemma/Finetune_with_CALM.ipynb:1]: Updated the docs now.

Comment (kinarr) [reply Gemma/Finetune_with_CALM.ipynb:1]: Done.

Comment (kinarr) [reply Gemma/Finetune_with_CALM.ipynb:1]: Happened to include it from the other paper and now I've removed it.

**Comment by kinarr** [comment]
@windmaple I've updated the docs and removed the Kaggle check now.

**Review by windmaple — APPROVED** [review]

**Pull request merged**